src/background/stores/global/user-configuration-store.ts

```diff
@@ -66,6 +71,14 @@ export class UserConfigurationStore extends BaseStoreImpl<UserConfigurationStore
 
     private onSetHighContrastMode = (payload: SetHighContrastModePayload): void => {
         this.state.enableHighContrast = payload.enableHighContrast;
+        this.state.lastSelectedHighContrast = payload.enableHighContrast;
+        this.saveAndEmitChanged();
+    };
+
+    private onSetNativeHighContrastMode = (payload: SetNativeHighContrastModePayload): void => {
+        this.state.enableHighContrast = payload.enableHighContrast
```

Comment:
can be swapped for:

`this.state.enableHighContrast = payload.enableHighContrast || this.state.lastSelectedHighContrast`

If you're concerned about it not existing or something like that (shouldn't happen under TS but regardless), you can use !! in front.

Reply:
I had it that way at first, but I kept having to look it over to convince myself it felt correct; I felt like this way was a little bit easier to read